src/v2/Apps/NewForYou/NewForYouApp.tsx

```diff
@@ -31,6 +22,7 @@ export const NewForYouApp: FC = () => {
         </Column>
       </GridColumns>
       <Spacer mt={4} />
+      <NewForYouArtworksGridQueryRenderer />
```

Comment:
No need to use a query renderer here, can just use the fragment container so that it properly renders on first load from the server. If there were below-the-fold areas, those we would use query renderers with with the `lazyLoad` prop attached.

Reply:
Yeah, for something like this we don't need to _only_ render on the client; user is logged in, it's all very deliberate content for one specific user in particular and so performance tuning here is something of a premature optimization. Maybe @dzucconi can comment more on his thinking? 

Can swap out this line with 
```tsx
<NewForYouArtworksGridFragmentContainer viewer={viewer} /> 
```
And achieve the same ends, no other change necessary, only there will no longer be a loading flash and the page will be stable on first render.

Reply:
Primarily the reason was to actually be able to debug the query because we had no insight into what the error was (MP  is swallowing error and apparently a malformed query can 500 one of the services (There is a bug there that should be fixed)). Moving it to the client-side allowed us to actually see what was happening.

Reply:
This MP swallowing errors business that we're seeing all over lately is driving me crazy. I'm going to open a ticket that we can add to as we see them. 

And also, we should look into adding a flag that we can toggle to force client side rendering. Relay will fetch there if a short circuit is thrown into the rehydration data that is mounted on the window, even if there's been a server-side pass. I'll also ticket that.

Reply:
Anything that y'all see going forward, please add here: https://artsyproduct.atlassian.net/jira/software/projects/WP/boards/97?issueParent=40613

Reply:
It's less that MP is swallowing errors and more that errors aren't being handled

Reply:
Updated the epic description slightly 👍